### PR TITLE
Implemented QDM Patient to HDS Record conversions

### PR DESCRIPTION
**Pull request by holmesie**
This PR includes an implementation of a module/class that converts the new QDM Patient model to the old HDS based Record model. See changes to the README.md for info on how to use it.

---

Pull requests into cqm-converter require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/TACO-54
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If applicable, the library version number in `cqm-converter.gemspec` has been updated

**Reviewer 1:**

Name: @okeefm 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @ljades 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 3:**

Name: @jbradl11 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: lib/cqm/converter/hds_record.rb [hunk 1]
```diff
@@ -77,28 +78,17 @@ def to_qdm(record)
         patient.data_elements << QDM::PatientCharacteristicExpired.new(expired_datetime: expired_datetime)
       end
 
-      # Convert patient characteristic payer.
```
Comment (okeefm) [inline lib/cqm/converter/hds_record.rb]: Does this bit no longer need to be converted?

Comment (holmesie) [reply lib/cqm/converter/hds_record.rb]: No, I didn't think it was one to one enough, so I moved it to extended data.

File: lib/cqm/converter/qdm_patient.rb [hunk 2]
```diff
@@ -2,9 +2,318 @@
 module CQM::Converter
   # CQM Converter class for QDM based patients.
   class QDMPatient
+    # Initialize a new HDSRecord converter. NOTE: This should be done once, and then
```
Comment (okeefm) [inline lib/cqm/converter/qdm_patient.rb]: Should this say `QDMPatient` converter, rather than HDSRecord?

Comment (holmesie) [reply lib/cqm/converter/qdm_patient.rb]: Indeed it should, fixed.

File: lib/cqm/converter/hds_record.rb [hunk 3]
```diff
@@ -108,6 +98,22 @@ def to_qdm(record)
       patient.family_name = record.last if record.last
       patient.bundle_id = record.bundle_id if record.bundle_id
 
+      # Convert extended_data.
```
Comment (okeefm) [inline lib/cqm/converter/hds_record.rb]: Bonnie team (@jbradl11 or whoever actually reviews this): the relevance of these `extended_data` attributes to Bonnie is something we should look at. If there are any that are useful to you as well as Cypress, we should pull them out into the core `Patient` object.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -3,10 +3,67 @@
...
+    self
+  end
+end
+
+# Extension to Hash to deep clean hashes for test comparison purposes. This makes sure
+# things are always in order, specifically for some tricky cases with data elements.
+#
+# This makes: {a: 'b', c: 'd'} == {c: 'd', a: 'b'}
+class Hash
+  def clean_hash(&block)
```
Comment (okeefm) [inline spec/spec_helper.rb]: If `clean_hash` is acting on the object it's called on, by Ruby-logic should it be named `clean_hash!`? Like `deep_reject_key!` above

File: lib/cqm/converter/hds_record.rb [hunk 5]
```diff
@@ -108,6 +98,22 @@ def to_qdm(record)
...
+      patient.extended_data = {}
+      patient.extended_data['type'] = record.type if record.respond_to?('type')
+      patient.extended_data['measure_ids'] = record.measure_ids if record.respond_to?('measure_ids')
+      patient.extended_data['source_data_criteria'] = record.source_data_criteria if record.respond_to?('source_data_criteria')
+      patient.extended_data['expected_values'] = record.expected_values if record.respond_to?('expected_values')
```
Comment (jbradl11) [inline lib/cqm/converter/hds_record.rb]: This line may need to be modified based on the final design of the new "Expected Values" work, because Expected_values will most likely exist on the patient model and not in the extended_data.

Comment (edeyoung) [reply lib/cqm/converter/hds_record.rb]: i'm thinking along the lines of @holmesie
